docs/json.md

```diff
@@ -0,0 +1,421 @@
+Also, API users don't need to be concerned with the fact that an ID is a numeric value internally.
+
+Theoretically, changing the type of entity IDs could be extremely disruptive when API 4.0 is released.
+
+We have worked on minimizing this disruption for all Looker-provided language SDKs. This document describes some JSON parsing requirements and how our SDKs support them.
+
+## JSON parsing requirements
+
+There are many requirements for parsing and producing JSON, but most of the issues are identical among languages that support JSON **serialization** and **deserialization**.
+Other names for this process are **marshalling** and **unmarshalling** or **encoding** and **decoding**, but they all mean basically the same thing.
+
+Serialization is the conversion of a JSON payload into a structure, record, or object of a given language. Deserialization is rendering an instantiated data type to its JSON representation.
+
+Almost all languages include JSON serialization and deserialization support. Many include more than one, with an offering from a third-party as the de facto standard, rather than the first-party solution provided by the language vendor.
+
+The parsing requirements of concern for the Looker language SDKs are:
+
+1. JSON value types conflicting with the API type specification. This is called **type ambiguity** below. For the Looker API, these conflicts should be automatically resolvable for our SDKs. The conflict is either:
+   1. a numeric value for a string property (forward compatibility issue)
+   2. a string value for an integer property (backward compatibility issue)
+2. New properties in a JSON payload. These are ignored by the SDKs. (See [TypeScript caveats](#typescript-caveats) for additional comments.)
+3. Removed properties in a JSON payload. Typically, the property will end up unassigned. Some SDKs may throw errors with missing required properties, but that client code should be changed anyway if the property no longer exists.
+4. Name changes, like spaces to folders and homepages to boards. There is no reasonable automatic solution for this, so type or property renames will require changing SDK existing code.
+
+### forward and backward compatibility
+
+In this document, **forward compatible** means an older SDK can process payloads from a newer Looker API server.
+
+**Backward compatible** means a newer SDK can process payloads from an older Looker API server.
+
+Our goal with the Looker language SDKs is to provide both forward and backward compatibility.
+
+The current Looker API server is written in Ruby, which is one of the aforementioned "flexibly typed" languages. The Looker API server accepts ID value references as **either** `integer` or `string` values and converts them internally to the required type.
+This means the LookerAPI server is fully compatible with any Looker SDK for identically named properties and parameters that are integer in an older version and string in a newer version.
+
```

Comment:
```
In this document, **forward compatible** means an older SDK can deserialize payloads from a newer API specification, and  **backward compatible** means a newer SDK can deserialize payloads from an older API spec. The Looker SDK JSON deserializing requirements provide limited forward and backward compatibility when deserializing JSON payloads.

Outside of Looker SDK deserialization, Looker SDK serialized request payloads are fully compatible with Looker instance. Looker instance accepts ID as **either** `integer` or `string` type and converts them internally to the required type for identically named properties and parameters.
```
Removed mention of Ruby, because I think exposing the language or implementation details of a Looker instance is outside the scope of this doc.